### PR TITLE
Raise AirflowException when a subplan fails

### PR DESCRIPTION
**Pull request by mgasner**

### CODE REVIEW

**Comment by mgasner** [comment]
No special semantics for AirflowError -- it's just what Airflow operators raise when something goes wrong.

File: python_modules/dagster-airflow/dagster_airflow_tests/conftest.py [hunk 1]
```diff
@@ -211,3 +214,61 @@ def scaffold_dag(airflow_test):
...
+        execution_plan,
+        execution_date,
+        os.path.abspath(os.path.join(dags_path, os.path.basename(static_path))),
+        os.path.abspath(os.path.join(dags_path, os.path.basename(editable_path))),
+    )
+
+    os.remove(os.path.abspath(os.path.join(dags_path, os.path.basename(static_path))))
+    os.remove(os.path.abspath(os.path.join(dags_path, os.path.basename(editable_path))))
+
+    try:
```
Comment (schrockn) [inline python_modules/dagster-airflow/dagster_airflow_tests/conftest.py]: could you comment re: what is going on here. magic

File: python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py [hunk 2]
```diff
@@ -454,7 +486,17 @@ def execute(self, context):
             self._run_id = context['dag_run'].run_id
         try:
             # FIXME implement intermediate result persistence to S3
-            return super(DagsterOperator, self).execute(context)
+
+            self.log.debug('Executing with query: {query}'.format(query=self.query))
+            raw_res = super(DagsterOperator, self).execute(context)
+            res = json.loads(raw_res)
+            if res['data']['startSubplanExecution']['hasFailures']:
+                raise AirflowException(
```
Comment (schrockn) [inline python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py]: shouldn't we be searching for the step that failed? there's no guarantee that it is first step no?

File: python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py [hunk 3]
```diff
@@ -454,7 +486,17 @@ def execute(self, context):
...
+                    res['data']['startSubplanExecution']['stepEvents'][0]['errorMessage']
+                )
+            return res
+        except JSONDecodeError:
```
Comment (schrockn) [inline python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py]: when does json error occur?

File: python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py (hunk 2 above)
Comment (schrockn) [reply python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py]: if there is only one step, i think adding a check.invariant would be good for documentation

Comment (mgasner) [reply python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py]: yep, this is improved in the immediately following PR

File: python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py (hunk 3 above)
Comment (mgasner) [reply python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py]: added a comment, but this is if e.g. dagit crashes

File: python_modules/dagster-airflow/dagster_airflow_tests/conftest.py (hunk 1 above)
Comment (mgasner) [reply python_modules/dagster-airflow/dagster_airflow_tests/conftest.py]: added comments on all the test fixtures

**Pull request merged**